src/components/sidebar/SideBar.tsx

```diff
@@ -0,0 +1,80 @@
+import { FC, useEffect, useState } from 'react';
+import { fetchCharacter } from '../../service/apiService';
+import { IApi } from '../../utils/types/types';
+import { useParams } from 'react-router-dom';
+import './sideBar.css';
+
+interface SideBarProps {
+  id: string;
+  isSideBarOpen: boolean;
+  closeSideBar: () => void;
+}
+
+const SideBar: FC<SideBarProps> = (props) => {
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [item, setItem] = useState<IApi>();
+  const [hasError, setHasError] = useState(false);
+  const { id } = useParams();
+
+  useEffect(() => {
+    if (id) {
+      getCharacter(id);
+    }
+  }, [id]);
+
+  const getCharacter = (id: string) => {
+    false;
+    setHasError(false);
+
+    fetchCharacter(id)
+      .then((data) => {
+        setIsLoaded(true);
+        setItem(data);
+        setHasError(false);
+      })
+      .catch((error) => {
+        console.error('Fetch error:', error);
+        setHasError(true);
+      });
+  };
+
+  if (!isLoaded) {
+    return <div>Loading...</div>;
```

Comment:
The similar case as in the ResultCatalog.tsx
```
 return (
    <>
      {!isLoaded && <div>Loading...</div>}
      {hasError ? (
        <div>Oops, nothing found!!!</div>
      ) : (
        <div className="section main-section">
          <h2>Serch results:</h2>
          ...
        </div>
      )}
    </>
  );
```

Reply:
Thank's!!!